### PR TITLE
feat: Added table component

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

This PR adds table component to ADS v2. Table component will be using ec-table internally.

Fixes https://github.com/appsmithorg/appsmith/issues/27771

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual on storybook 
- Manual on main repo
- Jest
- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: packages/design-system/src/Table/reset.css [hunk 1]
```diff
@@ -0,0 +1,55 @@
+.rc-table.ads-v2-table-wrapper {
```
Comment (KelvinOm) [inline packages/design-system/src/Table/reset.css:1]: Why not keep all styles in one place? You already have classes for all these overrides. I think it's better to move.

File: packages/design-system/src/Table/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    body: {
+      wrapper: StyledBody,
+      row: StyledRow,
+      cell: StyledCell,
+    },
+  };
+  return (
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
```
Comment (KelvinOm) [inline packages/design-system/src/Table/Table.tsx:39]: Could we fix this?

**Review by KelvinOm — COMMENTED** [review]
Something is wrong with the table header.
Снимок экрана 2023-10-24 в 18 06 43

File: packages/design-system/src/Table/Table.styles.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  className: clsx(TableHeaderRowClassName, className),
+}))`
+  height: var(--ads-v2-spaces-13);
+`;
+
+export const StyledHeaderCell = styled.th.attrs(({ className }) => ({
+  className: clsx(TableHeaderCellClassName, className),
+}))`
+  && {
+    font-size: 14px;
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply packages/design-system/src/Table/Table.styles.tsx:35]: Yeah font-size and weight have... Missed it. Thank you.

File: packages/design-system/src/Table/reset.css (hunk 1 above)
Comment (albinAppsmith) [reply packages/design-system/src/Table/reset.css:1]: So the reason for doing this here is, RC adds some divs on the fly and the classes we add is not applied on these. That's why I had to keep a seperate CSS file for this.

File: packages/design-system/src/Table/Table.tsx (hunk 2 above)
Comment (albinAppsmith) [reply packages/design-system/src/Table/Table.tsx]: Fixed it.

**Review by KelvinOm — DISMISSED** [review]

**Review by jsartisan — APPROVED** [review]

**Comment by kavitasmoolya** [comment]
@albinAppsmith 
Following are checked on DS repo [here](https://design-system-git-table-component-get-appsmith.vercel.app/?path=/story/design-system-table--table-story) and no issue found during testing

- Header height, { can ignore row height since it depends on content}
- Header color
- Header font styles
- Props: columns, data, sticky, emptyText
- border colors (every row have a border bottom)

**Pull request merged**